app/src/ui/delete-branch/delete-branch-dialog.tsx

```diff
@@ -95,14 +99,17 @@ export class DeleteBranch extends React.Component<
   }
 
   private deleteBranch = async () => {
-    const { dispatcher, repository, branch, onDeleted } = this.props
+    const { dispatcher, repository, branch } = this.props
+
+    this.setState({ isDeleting: true })
+
     await dispatcher.deleteBranch(
       repository,
       branch,
       this.state.includeRemoteBranch
     )
-    onDeleted(repository)
+    this.props.onDeleted(repository)
```

Comment:
This is going against the tidying-up goals of object destructuring earlier but I wanted to make this clear that we were calling a prop rather than _some other function_ (instance or otherwise).